clippy_lints/src/methods/mod.rs

```diff
@@ -4284,6 +4285,29 @@ declare_clippy_lint! {
     "map of a trivial closure (not dependent on parameter) over a range"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for `repeat().take()` that can be replaced with `repeat_n()`.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Using `repeat_n()` is more concise and clearer.
```

Comment:
May also be worth pointing out that it can be faster for types that are non-trivial to clone as it knows the repeat count and only needs to clone n-1 times